src/data-table/ColumnSelectorDialog.tsx

```diff
@@ -24,12 +26,33 @@ export function ColumnSelectorDialog<T extends ReferenceObject>(
         onChange,
         onCancel,
         allowReorderingColumns = true,
+        keepDisabledColumns = true,
     } = props;
-    const sortableColumns = columns.map(({ name, text: label, disabled }) => ({
-        label,
-        value: name,
-        disabled: disabled ?? false,
-    }));
+    const sortableColumns = columns.map(
+        ({ name, text: label, disabled }): TransferOption => ({
+            label,
+            value: name.toString(),
+            disabled: disabled ?? false,
+        })
+    );
+
+    const selectedColumns = React.useMemo((): string[] => {
+        const disableColumns = columns.filter(col => col.disabled).map(col => col.name.toString());
```

Comment:
In the useCallback we check keepDisabledColumns to build the final value, for the useMemo we can do the same, and now `selectedColumns` can be passed directly to selected.